tactics/equality.ml

```diff
@@ -1049,9 +1040,8 @@ let discr_positions env sigma { eq_data = (lbeq,(t,t1,t2)); eq_clenv = eq_clause
   in
     discriminator >>= fun discriminator ->
     discrimination_pf e (t,t1,t2) discriminator lbeq false_kind >>= fun pf ->
-    let pf_ty = mkArrow (clenv_type eq_clause) Sorts.Relevant false_0 in
-    let absurd_clause = apply_on_clause (pf,pf_ty) eq_clause in
-    let pf = Clenv.clenv_value_cast_meta absurd_clause in
```

Comment:
Are you saying that the call to `clenv_fchain` there is about matching two types which are already the same and thus useless?

Reply:
Yes, it's what the commit comment says.